services/src/main/scala/cromwell/services/cost/GcpCostCatalogService.scala

```diff
@@ -212,23 +235,47 @@ class GcpCostCatalogService(serviceConfig: Config, globalConfig: Config, service
     }
 
   // TODO consider caching this, answers won't change until we reload the SKUs
-  def calculateVmCostPerHour(instantiatedVmInfo: InstantiatedVmInfo): ErrorOr[BigDecimal] =
-    for {
+  def calculateVmCostPerHour(instantiatedVmInfo: InstantiatedVmInfo): ErrorOr[BigDecimal] = {
```

Comment:
This method had to be significantly restructured due to the fact that, unlike with CPUs and RAM, we need to be able to skip the GPU SKU lookup and cost calculation steps for a VM that does not use GPUs. Note that we can't just try to look up the SKU anyway and give the GPU cost as 0 if there is an error if we want to be able to distinguish between not having anything to look up and having a genuine error because, for example, the GPU type is not recognized.

To the best of my knowledge, this necessitated breaking up the large `for-yield` block into separate pieces. (It might not have been the most readable anyway if I added more steps to the already large `for-yield` that existed before.)